core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -138,7 +137,7 @@ public long getLength() {
       private final SimpleBufferedOutputStream fsBufferedOutput;
       private OutputStream out;
 
-      public WBlockState(Algorithm compressionAlgo, RateLimitedOutputStream fsOut,
+      public WBlockState(DefaultCompressionAlgorithm compressionAlgo, RateLimitedOutputStream fsOut,
```

Comment:
It looks like you replaced the enum with the default class in multiple places here. Would it be better to pass the interface type `CompressionAlgorithm` in these places? I am not sure if you want to always pass around the default class.

Reply:
`CompressionAlgorithm` is really just the config. `DefaultCompressionAlgorithm` is the one, and currently only, implementation. There is no ability for a user to supply their own algorithm implementation, just the parameters/config for the algorithm. Maybe I need to change the class names to make that more clear

Reply:
I renamed `CompressionAlgorithm` to `CompressionAlgorithmConfiguration` in 9b22ead